pkg/api/unversioned/types.go

```diff
@@ -339,6 +339,8 @@ type APIResource struct {
 	Name string `json:"name"`
 	// namespaced indicates if a resource is namespaced or not.
 	Namespaced bool `json:"namespaced"`
+	// kind is the kind for the resource (e.g. 'Foo' is the kind for a resource 'foo')
+	Kind string `json:"kind"`
```

Comment:
> API resource is served under a specific GroupVersion (e.g. /apis/company.com/v1) so it seems redundant.

Consider `replicationcontrollers/scale`.  In the end it should live in the "" group, but accept a kind from the "extensions" group.  Its a general problem we have with that sort of endpoint, since something like the HPA controller should be able to scale anything in any group.